test/test_torch.py

```diff
@@ -19094,6 +19095,22 @@ class TestTorchMathOps(TestCase):
 class TestTorch(AbstractTestCases._TestTorchMixin):
     exact_dtype = True
 
+# Tests that appropriate source range is highlighted in error messages.
+class TestHighlitedSource(unittest.TestCase):
+    def test_source_highlighted(self):
+        def to_list_missing_type_annotation(x):
+            # type: (torch.Tensor) -> List[float]
+            li = x.tolist()
+            return li
+
+        try:
+            m = torch.jit.script(to_list_missing_type_annotation)
+        except Exception as e:
+            FileCheck().check_source_highlighted("x.tolist").run(str(e))
+            return
+
+        self.assertFalse("Exception should have been raised earlier.")
```

Comment:
I think a better approach is to add something like `checkScriptRaisesRegexWithHighlight` to `JitTestCase` so that we can write something like:
```
        with self.assertRaisesRegexWithHighlight(
            RuntimeError, r"Expected type hint for result of tolist()", "tolist"
        ):
            self.checkScript(to_list_missing_type_annotation, (torch.randn(5),))
```

Reply:
Done.